src/Entity/FileEntity.php

```diff
@@ -163,7 +185,7 @@ public function postSave(EntityStorageInterface $storage, $update = TRUE) {
       if (\Drupal::moduleHandler()->moduleExists('image') && $this->getMimeTypeType() == 'image' && $this->getSize()) {
         // If the image dimensions have changed, update any image field references
         // to this file and flush image style derivatives.
-        if (isset($this->original->metadata['width']) && ($this->metadata['width'] != $this->original->metadata['width'] || $this->metadata['height'] != $this->original->metadata['height'])) {
+        if (isset($this->original->metadata['width']) && ($this->getMetadata('width') != $this->original->metadata['width'] || $this->getMetadata('height') != $this->original->metadata['height'])) {
```

Comment:
You can drop the isset() checks here now since the method will abstract that for us.